### PR TITLE
Two New UI Features

### PR DESCRIPTION
**Pull request by cassidysymons**
1) Adds support for a "test mode" system message that indicates that users should stay off the platform.
2) When a user logs sample information, it checks when they last updated their surveys - if it has been over a week, it prompts them to review their survey answers.

This PR should be merged after https://github.com/biocore/microsetta-interface/pull/274

### CODE REVIEW

**Comment by cassidysymons** [comment]
Leaving it as a draft for the moment as the Spanish translations for the new messages need to be dropped in on Monday morning.

File: microsetta_interface/implementation.py [hunk 1]
```diff
@@ -1631,13 +1678,25 @@ def get_nutrition(*, account_id=None, source_id=None):
     if has_error:
         return source_output
 
+    # Check the FFQ prereqs for the source (birth year, gender, height, weight
```
Comment (wasade) [inline microsetta_interface/implementation.py:1681]:
```suggestion
    # Check the FFQ prereqs for the source (birth year, gender, height, weight)
```

File: microsetta_interface/static/vue_survey_form_es_mx.js [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+            schema: survey_schema,
+            formOptions: {
+                validateAfterLoad: true,
+                validateAfterChanged: true
+            }
+        };
+    },
+
+    methods: {
+        prettyJSON: function(json) {
```
Comment (wasade) [inline microsetta_interface/static/vue_survey_form_es_mx.js:20]: A lot of the stuff here and below looks like it's replicated across multiple files. Can the common logic be centralized?

File: microsetta_interface/templates/kits.jinja2 [hunk 3]
```diff
@@ -129,7 +129,7 @@
             let kit_validation_setup_func = make_kit_validation_setup_func(form_name, kitSubmitHandler);
             kit_validation_setup_func()
 
-            {% for kit_samples in kits.values() | sort() %}
+            {% for kit_samples in kits.values() %}
```
Comment (wasade) [inline microsetta_interface/templates/kits.jinja2:132]: Was the sort not working?

File: microsetta_interface/templates/kits.jinja2 [hunk 4]
```diff
@@ -223,9 +228,18 @@
...
+                                <div class="barcode-col barcode-col-hide-mobile">
                                     <img src="/static/img/{{ sample.alert_icon }}" width="24" height="24" />
                                 </div>
+                                {% if sample.css_class == 'sample-complete' %}
+                                    <div class="barcode-col barcode-date-text barcode-col-hide-mobile">
+                                        {{ _('Collected') }}: {{ sample.sample_datetime }}
+                                    </div>
+                                {% else %}
+                                    <div class="barcode-col barcode-warning-text barcode-col-hide-mobile">
+                                        Info Needed
```
Comment (wasade) [inline microsetta_interface/templates/kits.jinja2:240]:
```suggestion
                                        {{ _('Info Needed') }}
```

File: microsetta_interface/templates/nutrition.jinja2 [hunk 5]
```diff
@@ -66,6 +70,16 @@
...
+
+            /*$('button').on('click', function() {
+                if($("form[name='ffq_code_form']").valid()) {
+                    updateButtonState("");
+                //    $('#ffq_code_button').prop('disabled', true);
+                } else {
+                    updateButtonState("yes");
+                //    $('#ffq_code_button').prop('disabled', false);
+                }
+            });*/
```
Comment (wasade) [inline microsetta_interface/templates/nutrition.jinja2:82]:
```suggestion
```

File: microsetta_interface/templates/source.jinja2 [hunk 6]
```diff
@@ -49,20 +51,24 @@
...
             <div class="col-6 col-md-4 p-2">
-                <div class="card card-survey">
+                <div class="card card-survey {% if detail.date_last_taken is none and detail.survey_template_id == 10 %}card-survey-highlight{% endif %}">
                     <div class="row m-2 survey-info-row">
-                        {% if detail.date_last_taken != None %}
+                        {% if detail.date_last_taken is not none %}
                             <div class="col-8 small-text survey-info">{{ _('Last modified') }} {{ detail.date_last_taken }}</div>
                         {% else %}
-                            <div class="col-8 card-survey-new">{{ _('NEW') }}</div>
+                            {% if detail.survey_template_id == 10 %}
```
Comment (wasade) [inline microsetta_interface/templates/source.jinja2:67]: Could `10` be replaced with a named variable at the top of the template?

File: microsetta_interface/translations/en_US/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -6083,3 +6188,57 @@ msgstr ""
...
+
+#~ msgid ""
+#~ "Lifestyle, health, and diet information "
+#~ "is essential in order to gain "
+#~ "novel insights into the human microbiome"
+#~ " and design better studies. You can"
+#~ " help us by completing our surveys,"
+#~ " starting with your Basic Information."
+#~ msgstr ""
+
```
Comment (wasade) [inline microsetta_interface/translations/en_US/LC_MESSAGES/messages.po:6244]:
```suggestion
```

**Review by wasade — CHANGES_REQUESTED** [review]
Thanks! Most comments are minor. It would I think be valuable to decompose the common logic in the `vue_survey_form_*.js` scripts though as the duplications represent a maintenance burden (and avenue for bugs) should that logic need to be modified in the future

**Pull request closed without merging**